birdhouse/deployment/triggerdeploy.sh

```diff
@@ -100,121 +100,125 @@ should_trigger() {
+    git show "${CURRENT_REMOTE_BRANCH}":./autodeploy/conditional-trigger > "${TMP_SCRIPT}"
     if [ "$?" -eq 0 ]; then
         # Execute script only if it exists, script is optional.
-        chmod a+x $TMP_SCRIPT
-        GIT_CHANGED_FILES="$GIT_CHANGED_FILES" $TMP_SCRIPT
+        chmod a+x "${TMP_SCRIPT}"
+        GIT_CHANGED_FILES="$GIT_CHANGED_FILES" "${TMP_SCRIPT}"
         CONDITIONAL_TRIGGER_EXIT_CODE=$?
     fi
-    rm $TMP_SCRIPT
+    rm "${TMP_SCRIPT}"
 
     # trigger deploy if the behind count is greater than zero and ahead count is
     # less than or equal zero
-    if [ "$BEHIND_CNT" -gt 0 ]; then
-        if [ "$AHEAD_CNT" -le 0 ]; then
-            if [ $CONDITIONAL_TRIGGER_EXIT_CODE -eq 0 ]; then
-                echo "triggerdeploy: repo '$EXTRA_REPO' is behind will trigger deploy"
+    if [ "${BEHIND_CNT}" -gt 0 ]; then
+        if [ "${AHEAD_CNT}" -le 0 ]; then
+            if [ ${CONDITIONAL_TRIGGER_EXIT_CODE} -eq 0 ]; then
+                echo "triggerdeploy: repo '${EXTRA_REPO}' is behind will trigger deploy"
                 return 0
             else
-                echo "triggerdeploy: repo '$EXTRA_REPO' is behind but do not need to trigger deploy, just pull if clean checkout"
+                echo "triggerdeploy: repo '${EXTRA_REPO}' is behind but do not need to trigger deploy, just pull if clean checkout"
                 # pull only if clean checkout to avoid overriding important changes
-                if [ -z "`git status -u --porcelain`" ]; then
+                if [ -z "$(git status -u --porcelain)" ]; then
                     git pull
                 else
-                    echo "triggerdeploy: WARNING: unclean repo '$EXTRA_REPO', unable to pull" 1>&2
+                    echo "triggerdeploy: WARNING: unclean repo '${EXTRA_REPO}', unable to pull" 1>&2
                 fi
                 return 1
             fi
         else
-            echo "triggerdeploy: do nothing, repo '$EXTRA_REPO' is both behind and ahead (divergent), must be a devel repo"
+            echo "triggerdeploy: do nothing, repo '${EXTRA_REPO}' is both behind and ahead (divergent), must be a devel repo"
             return 1
         fi
     else
-        echo "triggerdeploy: repo '$EXTRA_REPO' is up-to-date or ahead, no deploy needed"
+        echo "triggerdeploy: repo '${EXTRA_REPO}' is up-to-date or ahead, no deploy needed"
         return 1
     fi
 }
 
 
-START_TIME="`date -Isecond`"
+START_TIME="$(date -Isecond)"
 echo "==========
-triggerdeploy START_TIME=$START_TIME"
+triggerdeploy START_TIME=${START_TIME}"
 
-. "${COMPOSE_DIR}/scripts/read-configs.include.sh"
+. "${COMPOSE_DIR}/read-configs.include.sh"
 
 # Read AUTODEPLOY_EXTRA_REPOS
 read_basic_configs_only
 
 set -x
 
 SHOULD_TRIGGER=""
-for adir in $COMPOSE_DIR $AUTODEPLOY_EXTRA_REPOS; do
-    if [ -d "$adir" ]; then
-        cd $adir
+for adir in "${COMPOSE_DIR}" ${AUTODEPLOY_EXTRA_REPOS}; do
+    if [ -d "${adir}" ]; then
+        cd "${adir}" || exit
```

Comment:
Do not `exit` here because we allow non-existing dir in `AUTODEPLOY_EXTRA_REPOS` !

Reply:
`-d "${adir}"` just before is to ensure it exists
A failure to `cd` here would be an actual unexpected error (e.g.: bad permissions).